### PR TITLE
integration test for catalogue

### PR DESCRIPTION
**Pull request by Alvtag**

### CODE REVIEW

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 1]
```diff
@@ -46,14 +44,14 @@ public void flightSearchTest() throws Exception {
         Traveler.flightSearch(query, new FlightSearchCallback() {
             @Override
             public void onFlightSearchSuccess(List<Flight> flights) {
-                Log.v(TAG, "TravelerTest::onFlightSearchSuccess");
+                TravelerLog.v(TAG, "TravelerTest::onFlightSearchSuccess");
```
Comment (ata-n) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: Log

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 2]
```diff
@@ -62,4 +60,81 @@ public void onFlightSearchError(Error error) {
         Assert.assertEquals("Error(s) occurred while querying API for flight 'AC100'!\n" +
                 "See logcat for more info\n", 0, errorList.size());
     }
+
+    @Test
+    public void catalogueTest() throws Exception {
```
Comment (ata-n) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: incorrect spelling

Comment (ata-n) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: Also in commit messages, (just saying for the future, no need to reedit the messages, but the method needs to read correctly)

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 3]
```diff
@@ -62,4 +60,81 @@ public void onFlightSearchError(Error error) {
...
+        CountDownLatch latch = new CountDownLatch(1);
+        List<Flight> flights = new ArrayList<>();
+        final List<Error> errorList = new ArrayList<>();
```
Comment (ata-n) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: Again, error list is not a great way to track expectations

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 4]
```diff
@@ -62,4 +60,81 @@ public void onFlightSearchError(Error error) {
...
+     */
+    @Test
+    public void flightSearchWithCatalogueTest() throws Exception {
+        Date date = Calendar.getInstance().getTime();
+        FlightQuery query = new FlightQuery("AC100", date);
+
+        CountDownLatch latch = new CountDownLatch(1);
+        final List<Error> errorList = new ArrayList<>();
+
+        Traveler.flightSearch(query, new FlightSearchCallback() {
```
Comment (ata-n) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: No need for this chained calls, should create a mock Flight on the spot and use that

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java [hunk 5]
```diff
@@ -64,7 +63,8 @@ protected void main() {
         BlockTask sessionBeginBlockTask = new BlockTask() {
             @Override
             protected void main() {
-                if (!TextUtils.isEmpty(sessionBeginTask.getSession().getToken().getValue())) {
+                Token token = sessionBeginTask.getSession().getToken();
+                if (null != token && !TextUtils.isEmpty(token.getValue())) {
```
Comment (ata-n) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java]: Did you reach a null pointer exception here?

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java (hunk 4 above)
Comment (Alvtag) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: A mocked flight would need a valid flight ID, The flight IDs used to call the API are constantly changing, and Brian's data cannot query the past.  lets chat about this one

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java (hunk 1 above)
Comment (Alvtag) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: updated

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java (hunk 5 above)
Comment (Alvtag) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java]: I did, in the past. today I ran this code without the null check and it seems to work fine- I will remove this change

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java (hunk 2 above)
Comment (Alvtag) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: heh, didn't catch my canadian spelling there. fixed.

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java (hunk 4 above)
Comment (Alvtag) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: updated comment

**Pull request merged**